rfcs/0019-bundle-info-preload.md

```diff
@@ -0,0 +1,117 @@
+- Start Date: 2025-11-17
+- RFC PR: [#1194](https://github.com/UI5/cli/pull/1194)
+- Issue: -
+- Affected components <!-- Check affected components by writing an "X" into the brackets -->
+    + [x] [ui5-builder](./packages/builder)
+    + [ ] [ui5-server](./packages/server)
+    + [ ] [ui5-cli](./packages/cli)
+    + [ ] [ui5-fs](./packages/fs)
+    + [ ] [ui5-project](./packages/project)
+    + [ ] [ui5-logger](./packages/logger)
+
+
+# RFC 0019 Bundle Info Preload
+
+## Summary
+
+This RFC proposes a new bundling approach for UI5 libraries called "Bundle-Info Preload". Instead of a single monolithic `library-preload.js`, each library will be split into two bundles:
+
+1. `library-preload.js`: A small initial bundle containing the `library.js`, its direct dependencies, as well as "bundle-info" describing the content of the second bundle
+2. `_library-content.js`: A larger secondary bundle containing the remaining modules of the library.
+
+This allows the UI5 runtime to always load the small `library-preload.js` for all declared library dependencies, including those marked as "lazy". The larger `_library-content.js` is only loaded on-demand when a module from within it is actually requested. This new mechanism eliminates the need for application developers to manually trigger the loading of lazy dependencies.
+
+## Motivation
+
+Currently, UI5 applications often declare library dependencies as "lazy" in their `manifest.json` to improve initial loading times. However, this requires developers to explicitly load these libraries using `sap/ui/core/Lib#load("sap.m")` (or the deprecated variant `sap.ui.getCore().loadLibrary("sap.m")`) before any of their modules can be used. This is a manual and error-prone process.
```

Comment:
It's `Lib.load` (static), and the name is passed via an options parameter. Maybe we should also link to the API Reference? https://ui5.sap.com/#/api/sap.ui.core.Lib%23methods/sap.ui.core.Lib.load.

I would rather prefer to not explicitly mention the deprecated variant, but I'm also ok with keeping it.